src/System.CommandLine.Tests/ArgumentTests.cs

```diff
@@ -442,6 +442,52 @@ public void Multiple_command_arguments_can_have_custom_parse_delegates()
                       .Contain("UH UH");
             }
 
+            [Theory]
+            // The two different examples verify that the relative order of the symbols doesn't matter.
+            [InlineData("--option-with-error 123 --depends-on-option-with-error")]
+            [InlineData("--depends-on-option-with-error --option-with-error 123")]
+            public void Custom_parser_can_check_another_option_result_for_custom_errors(string commandLine)
+            {
+                var optionWithError = new CliOption<string>("--option-with-error")
```

Comment:
subjective nit: by using the new C# syntax we can save space and avoid the need of using vars (if we ever move to dotnet/runtime there are some strict var usage guidelines)

```suggestion
                CliOption<string> optionWithError = new("--option-with-error")
```